### DISCUSSION TITLE
No Dialogic Class in Dialogic 2.0

### DISCUSSION THREAD

**Discussion by mechPenSketch** [dialogic-godot/dialogic#1483] [Q&A]
When I upgraded my Dialogic to 2.0, I got the following error in my output:
```
  res://addons/dialogic/Events/Text/event_text.gd:91 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Text/event_text.gd:141 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Other/DialogicGameHandler.gd:513 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Editor/directory_holder.gd:19 - Invalid call. Nonexistent function 'new' in base 'GDScript'.
  res://addons/dialogic/Events/Character/event_character.gd:88 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:90 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:117 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:119 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:142 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:144 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:164 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:166 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:252 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:299 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Editor/directory_holder.gd:35 - Invalid call. Nonexistent function 'new' in base 'GDScript'.
```

Mainly, that there is a missing `Dialogic` class.
For now, it does not affect my project in any other way, and I don't really have a way to test out my Dialogic on my project. Therefore, I don't have an issue to report. Should this be concerning, though?

Could it be that the `Dialogic` class script is missing, or has it been renamed to another class?

**Comment by Jowan-Spooner** [comment]
Did this persist after enabeling the plugin and restarting godot? This is typically a problem with godot trying to scan the scripts event thought the Dialogic singleton is only added when the plugin is enabled. It goes away once godot starts with the plugin already enabled.

**Reply by mechPenSketch** [reply]
Yes, the error is still being output even after restarting Godot with the plugin enabled. I have also disable dialogic plugin first, then enable it. It returned the following error message:

```
Can't add Autoload:
Invalid name.
Must not collide with an existing global script class name.
```

**Reply by mechPenSketch** [reply]
Maybe by reading `Dialogic`, godot is expect it to be a class when it's really a Singleton?

**Comment by mechPenSketch** [comment]
Is `Dialogic` the name of a Global autoload and not `class_name`? I initially thought it's a `class_name`, but after reading through the plugin code, it seems to be a singleton to be added to the autoload:
```
func _enable_plugin():
add_autoload_singleton("Dialogic", "res://addons/dialogic/Other/DialogicGameHandler.gd")
add_dialogic_default_action()
```
I've tried adding it to autoload manually, but I still get an error message saying the name would clash with an existing class `Dialogic`.

Apparantly, there's supposed to be a class script named `Dialogic`, but when I opened `res://addons/dialogic/Other/DialogicGameHandler.gd`, it has the output error stating missing `Dialogic` script.

For now, my solution is to rename the singleton differently.

**Reply by mechPenSketch** [reply]
Since `DialogicGameHandler.gd` is supposed to be the `Dialogic` singleton and its method `start(timeline, single_instance = true)` calls `Dialogic` for `start_timeline(timeline)`, in which it also has, maybe it can cut `Dialogic` off and have it call its own method `start_timeline(timeline)` instead.

Can the same be said for other scripts calling `Dialogic`?

`event_text`'s `from_text(string:String)` calls `Dialogic` for `character_directory` in which the singleton has, so I can't modify it.

**Comment by mechPenSketch** [comment]
In the end, I have edited `project.godot` with a text editor to add `Dialogic` into autoload. I still get the output error despite `Dialogic` being enabled. Maybe in GdScript 2.0, global singletons are called differently from classes?

For reference here is the out message:
```
  res://addons/dialogic/Events/Text/event_text.gd:91 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Text/event_text.gd:141 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Editor/directory_holder.gd:19 - Invalid call. Nonexistent function 'new' in base 'GDScript'.
  res://addons/dialogic/Events/Character/event_character.gd:88 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:90 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:117 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:119 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:142 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:144 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:164 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:166 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:252 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Events/Character/event_character.gd:299 - Parse Error: Could not find script for class "Dialogic".
  res://addons/dialogic/Editor/directory_holder.gd:35 - Invalid call. Nonexistent function 'new' in base 'GDScript'.
  Script does not inherit from Node: res://addons/dialogic/Other/DialogicGameHandler.gd.
  editor/editor_autoload_settings.cpp:551 - Condition "!info->node" is true. Continuing.
  modules/gdscript/gdscript.cpp:1981 - Condition "!named_globals.has(p_name)" is true.
  Script does not inherit from Node: res://addons/dialogic/Other/DialogicGameHandler.gd.
  editor/editor_autoload_settings.cpp:551 - Condition "!info->node" is true. Continuing.
```

**Reply by mechPenSketch** [reply]
According to [Godot's Singleton page](https://docs.godotengine.org/en/stable/tutorials/scripting/singletons_autoload.html), they can still be called by their name.

**Reply by mechPenSketch** [reply]
Seems that Godot registers `Dialogic` as a missing class before going to check the list of singletons in autoload. If that's the case, the problem could lie in the godot engine and not in this asset.

**Comment by mechPenSketch** [comment]
Okay, after experimenting with the plugin, I have diagnosed what's wrong with it and have made a proper issue here: #1489

**Comment by ConstantChange** [comment]
Also having this issue - parse error cant find classes = Alpha 13

class_name DialogicGameHandler

![image](https://github.com/dialogic-godot/dialogic/assets/72226015/00f2bd43-b768-459d-8aeb-0bea584c3ce5)

**Comment by Jowan-Spooner** [comment]
@ConstantChange Here's some things you can try:
- Delete the .godot folder to trigger recalculation of the global class cache
- Make sure you don't have multiple versions installed that might be clashing. When installing always properly remove the old version first.
